### DISCUSSION TITLE
Is it possible for stackstorm to work with Golang?

### DISCUSSION THREAD

**Discussion by MarlonHeiber** [StackStorm/st2#5566] [General]
Good Morning!
I'm new on Stackstorm, i did some research and didn't find much content, so I'll ask here in the community.

Is it possible for StackStorm to run an application in golang instead of Python?

I don't know if anyone here has tried this.

For example an Action that runs a Golang application that returns true or false on success or failure, the same as is done in Python today.

**Comment by arm4b** [comment]
Yes,
In short, your Golang app should just return a json-formatted result which will be parsed by stackstorm automatically. Non-zero exit code from the app will be considered as a failure.

See an example: [Turning Java App Into StackStorm Action](https://stackstorm.com/2015/09/15/java-action/)

**Comment by minsis** [comment]
To extend on @armab's answer.

I depends on what you're trying to do. You can go the route as above and just simply have your Golang app dump serialized data which StackStorm pack reads when dealing with something like core.remote, or core.local.

If you're trying to do something specific where you build a pack that interacts with a Golang app directly you'll have to do a little work in both Python and Golang to make it happen. In Golang you compile your app with the shared library and Python ctypes allows you to load that shared library and execute functions in it.
